### DISCUSSION TITLE
Star() object with epoch parameter seems to have no effect

### DISCUSSION THREAD

**Discussion by muratisik-lab** [skyfielders/python-skyfield#771] [Q&A]
Dear folks,

Thank you for this great package called Skyfield. I am using it to develop code that controls my DIY DSLR point-and-shoot platform with tracking capability.

I would like to clarify one specific point. When I use the Star() object with different epoch settings, they turn out exactly the same coordinates each time, as follows:

        from skyfield.api import N,S,E,W, wgs84, Star, load, Angle, utc
        
        planets = load('de421.bsp')
        earth = planets['earth']
        ts = load.timescale()
        t = ts.utc(2022, 8, 2, 13, 22)
        location = earth + wgs84.latlon( 41.033055 * N, 29.0 * E )
        
        teststar_ICRS = Star(ra=Angle(degrees=0.0), dec_degrees=90.0)
        teststar_Jnow = Star(ra=Angle(degrees=0.0), dec_degrees=90.0, epoch=ts.now())
        
        for star in (teststar_ICRS, teststar_Jnow):
            astrometric = location.at(t).observe(star)
            ra, dec, *_ = astrometric.radec()
            print("Ra/Dec: ", ra, dec)
            alt, az, *_ = astrometric.apparent().altaz()
            print("Alt/Az:", alt, az, "\n")
        
        OUTPUT:
        Ra/Dec:  08h 45m 52.71s +90deg 00' 00.0"
        Alt/Az: 41deg 09' 11.6" 359deg 59' 47.7" 
        
        Ra/Dec:  08h 45m 52.71s +90deg 00' 00.0"
        Alt/Az: 41deg 09' 11.6" 359deg 59' 47.7" 

I would expect a separation on the order of 0.12 degrees between them as of 2022.6.
Is my thinking valid? Appreciate your comments.

Thank you
Murat Isik

**Comment by brandon-rhodes** [comment]
Good question! The ‘epoch’ is the moment-in-time of the star position, not the ‘equinox’ of the coordinates—the coordinates are always in the ICRS fixed frame of reference. So the epoch only matters if you provide the star with a velocity, in which case the epoch names the exact moment at which the star was in the given position, with its velocity carrying it past that position if t > epoch.

I'll see if I can improve the docs this evening to make the distinction clear!

**Reply by AstroKeith** [reply]
I have the same issue. I think I understand the explanation but it leaves a gap for me.
I am measuring RA & Dec with a telescope, and hence they are ts.now(). I wish to compute the current AltAz of the objects, and it would also be convenient to have them as Skyfield objects so that other methods are available.(I do have other objects that are of ICRS frame of reference.)
Is there a way to handle the 'now' objects?
Thanks
Keith

**Reply by brandon-rhodes** [reply]
@AstroKeith, take a look at the `position_of_radec()` routine. Here are some links:

https://rhodesmill.org/skyfield/api-position.html#skyfield.positionlib.position_of_radec

https://rhodesmill.org/skyfield/examples.html#what-latitude-and-longitude-is-beneath-this-right-ascension-and-declination

The routine takes an argument `epoch=` that I think will do what you need: it makes the routine interpret the RA and Dec as dynamic coordinates referenced to the real Earth equator and poles of the given date. Let me know if it works for you!

**Comment by muratisik-lab** [comment]
Thanks Brandon,

position_of_radec() is just what I was asking for. It returns an ICRS object.

It is not possible to observe() it, though. How can I translate this position to an apparent Alt/Az coordinate pair with respect to my location?

To instantiate the question, for example, how do I calculate the horizontal coordinates of the celestial north point as of Jnow, from my location tonight?
